pkg/middleware/middleware_test.go

```diff
@@ -0,0 +1,46 @@
+package middleware
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+func TestExpectQueryParam(t *testing.T) {
+	t.Run("foo=bar", func(t *testing.T) {
+		m := ExpectQueryParam("foo")
+		r, _ := http.NewRequest("GET", "http://example.com?foo=bar", nil)
+		w := httptest.NewRecorder()
+		if m(w, r) != r {
+			t.Error("ExpectQueryParam did not recognize foo was provided")
+		}
+	})
+
+	t.Run("foo=", func(t *testing.T) {
+		m := ExpectQueryParam("foo")
+		r, _ := http.NewRequest("GET", "http://example.com?foo=", nil)
+		w := httptest.NewRecorder()
+		if m(w, r) != r {
+			t.Error("ExpectQueryParam did not recognize foo was provided")
+		}
+	})
+
+	t.Run("foo without equals sign", func(t *testing.T) {
+		m := ExpectQueryParam("foo")
+		r, _ := http.NewRequest("GET", "http://example.com?foo", nil)
+		w := httptest.NewRecorder()
+		if m(w, r) != r {
```

Comment:
Just want to make sure I understand the use of this test; if foo should not be recognized here, should the check be for `nil` and not for a validated request?

Reply:
`URL.Query()` considers `foo` to be present yet empty in this case and I figured `ExpectQueryParam()` should be consistent with the std lib:
```go
u, _ := url.Parse("google.com?foo=bar")
fmt.Println(u.Query())
u, _ = url.Parse("google.com?foo=")
fmt.Println(u.Query())
u, _ = url.Parse("google.com?foo")
fmt.Println(u.Query())
```
```
map[foo:[bar]]
map[foo:[]]
map[foo:[]]
```

Reply:
Gotcha! Thanks for the clarification!